src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -12592,5 +12592,1304 @@ public void ScopedReserved_Alias_Escaped()
+            var sourceA =
+@"using System.Diagnostics.CodeAnalysis;
+public class A<T>
+{
+    public ref T F1A(out T t1)
+    {
+        throw null;
+    }
+    public ref T F2A(scoped out T t2)
+    {
+        throw null;
+    }
+    public ref T F3A([UnscopedRef] out T t3)
+    {
+        t3 = default;
+        return ref t3;
+    }
+    public ref T F4A([UnscopedRef] scoped out T t4)
+    {
+        t4 = default;
+        return ref t4;
+    }
+}";
+            var comp = CreateCompilation(new[] { sourceA, UnscopedRefAttributeDefinition });
+            comp.VerifyEmitDiagnostics();
+            var refA = AsReference(comp, useCompilationReference);
+
+            var sourceB =
+@"class B : A<int>
+{
+    ref int F1B()
+    {
+        int i = 1;
+        return ref F1A(out i);
+    }
+    ref int F2B()
+    {
+        int i = 2;
+        return ref F2A(out i);
+    }
+    ref int F3B()
+    {
+        int i = 3;
+        return ref F3A(out i); // 1
+    }
+    ref int F4B()
+    {
+        int i = 4;
+        return ref F4A(out i); // 2
+    }
+}";
+            comp = CreateCompilation(sourceB, references: new[] { refA });
+            // https://github.com/dotnet/roslyn/issues/62791: Missing errors.
+            comp.VerifyEmitDiagnostics();
+
+            var baseType = comp.GetMember<NamedTypeSymbol>("B").BaseTypeNoUseSiteDiagnostics;
+            VerifyParameterSymbol(baseType.GetMethod("F1A").Parameters[0], "out System.Int32 t1", RefKind.Out, DeclarationScope.RefScoped);
+            VerifyParameterSymbol(baseType.GetMethod("F2A").Parameters[0], "out System.Int32 t2", RefKind.Out, DeclarationScope.RefScoped);
+            VerifyParameterSymbol(baseType.GetMethod("F3A").Parameters[0], "out System.Int32 t3", RefKind.Out, DeclarationScope.Unscoped);
+            VerifyParameterSymbol(baseType.GetMethod("F4A").Parameters[0], "out System.Int32 t4", RefKind.Out, DeclarationScope.Unscoped);
+        }
+
+        [Fact]
+        public void UnscopedRefAttribute_RefParameter()
+        {
+            var source =
+@"using System.Diagnostics.CodeAnalysis;
+public class A<T>
+{
+    public ref T F1A(ref T t1) => ref t1;
+    public ref T F2A(scoped ref T t2) => throw null;
+    public ref T F3A([UnscopedRef] ref T t3) => ref t3;
+    public ref T F4A([UnscopedRef] scoped ref T t4) => ref t4;
+}
+class B : A<int>
+{
+    ref int F1B()
+    {
+        int i = 1;
+        return ref F1A(ref i); // 1
+    }
+    ref int F2B()
+    {
+        int i = 2;
+        return ref F2A(ref i);
+    }
+    ref int F3B()
+    {
+        int i = 3;
+        return ref F3A(ref i); // 2
+    }
+    ref int F4B()
+    {
+        int i = 4;
+        return ref F4A(ref i); // 3
+    }
+}";
+            var comp = CreateCompilation(new[] { source, UnscopedRefAttributeDefinition });
+            comp.VerifyEmitDiagnostics(
+                // (6,23): error CS9064: UnscopedRefAttribute can only be applied to parameters that are passed by reference and implicitly scoped.
```

Comment:
>UnscopedRefAttribute can only be applied to parameters that are passed by reference and implicitly scoped. [](http://example.com/codeflow?start=41&length=106)

Consider "UnscopedRefAttribute can only be applied to implicitly-scoped 'out' parameters"?

Reply:
UnscopedRefAttribute can also be applied to `ref` parameters of `ref struct` types.

Reply:
Updated the text to cover the two cases as we discussed.